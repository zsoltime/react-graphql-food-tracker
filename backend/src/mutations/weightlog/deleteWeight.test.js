'use strict';

const deleteWeight = require('./deleteWeight');

describe('deleteWeight mutation', () => {
  it('deletes and returns the weightLog entry', async () => {
    const args = {
      id: 'validFoodId',
    };
    const mockedContext = {
      db: {
        mutation: {
          deleteWeightLog: jest.fn().mockImplementation(() => Promise.resolve({
            ...args,
            date: new Date('2018-11-20'),
            image: null,
            weight: 88.8,
          })),
        },
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([{ id: 'validUserId' }])),
        },
      },
      request: { userId: 'validUserId' },
    };

    const weight = await deleteWeight(null, args, mockedContext);

    expect(mockedContext.db.query.users).toBeCalled();
    expect(
      mockedContext.db.query.users.mock.calls[0]
    ).toMatchSnapshot();
    expect(mockedContext.db.mutation.deleteWeightLog).toBeCalled();
    expect(
      mockedContext.db.mutation.deleteWeightLog.mock.calls[0]
    ).toMatchSnapshot();
    expect(weight).toMatchSnapshot();
  });

  it('throws error if user is not signed in', () => {
    const mockedContext = {
      request: {},
    };

    expect(
      deleteWeight(null, null, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });

  it('throws error if ID is not valid', () => {
    const args = {
      id: 'invalidFoodId',
    };
    const mockedContext = {
      db: {
        query: {
          users: jest
            .fn()
            .mockImplementation(() => Promise.resolve([])),
        },
      },
      request: { userId: 'validUserId' },
    };

    expect(
      deleteWeight(null, args, mockedContext)
    ).rejects.toThrowErrorMatchingSnapshot();
  });
});
