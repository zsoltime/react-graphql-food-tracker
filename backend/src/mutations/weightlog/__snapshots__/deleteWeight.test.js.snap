// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteWeight mutation deletes and returns the weightLog entry 1`] = `
Array [
  Object {
    "where": Object {
      "AND": Array [
        Object {
          "id": "validUserId",
          "weightLogs_some": Object {
            "id": "validFoodId",
          },
        },
      ],
    },
  },
]
`;

exports[`deleteWeight mutation deletes and returns the weightLog entry 2`] = `
Array [
  Object {
    "where": Object {
      "id": "validFoodId",
    },
  },
]
`;

exports[`deleteWeight mutation deletes and returns the weightLog entry 3`] = `
Object {
  "date": 2018-11-20T00:00:00.000Z,
  "id": "validFoodId",
  "image": null,
  "weight": 88.8,
}
`;

exports[`deleteWeight mutation throws error if ID is not valid 1`] = `"You don't have permission to delete this entry"`;

exports[`deleteWeight mutation throws error if user is not signed in 1`] = `"You must be signed in to delete this entry"`;
